test/backports/BackportIteratorTest.cpp

```diff
@@ -0,0 +1,82 @@
+// Copyright 2025 The QLever Authors, in particular:
+//
+// 2025 Johannes Kalmbach <kalmbach@cs.uni-freiburg.de>, UFR/QL
+//
+// UFR = University of Freiburg, Chair of Algorithms and Data Structures
+// QL =  QLeverize AG
+
+#include <gmock/gmock.h>
+
+#include <forward_list>
+#include <vector>
+
+#include "backports/iterator.h"
+#include "util/ResetWhenMoved.h"
+
+namespace {
+
+using R = ad_utility::ResetWhenMoved<bool, false>;
+// A value that is initially `true` and becomes `false` when being moved.
+R r{true};
+
+// A value that is false and will stay false (used for comparison).
+const R rFalse{false};
+
+// The test suite is templated on a container that stores elements of type `R`
+// (see above).
+template <typename T>
+struct MoveSentinelTest : public ::testing::Test {
+ public:
+  // Initial state, 10 unmoved elements.
+  T container = []() { return T(10, r); }();
+
+  // Expected state after elementwise moving of the `container`.
+  T afterMove = []() { return T(10, rFalse); }();
```

Comment:
```suggestion
  T afterMoveT(10, rFalse);
```

What about this, doesn't this just work? (Same for the one above)

Reply:
That one doesn't work (it declares a function). 
T afterMove{10, rFalse) also doesn't work. 
But `T afterMove = T(10, rFalse)` does, so I updated to that.